Jenkinsfile

```diff
@@ -208,12 +210,22 @@ def job = {
                                 sh "docker pull ${config.dockerRegistry}${dockerRepo}:${config.cp_version}-latest"
                             }
 
+                            // We need to replace the parent version range before we can run any maven commands
+                            def pomFile = readFile('pom.xml')
+                            def parentVersionPattern = Pattern.compile(/(.*<parent>.*<groupId>io.confluent\S*<\/groupId>.*<version>)\[\d+\.\d+\.\d+-\d+\,\s+\d+\.\d+\.\d+-\d+\)(<\/version>.*<\/parent>.*)/, Pattern.DOTALL)
+                            // Groovy regex replaces the groups we didn't match, so we print the beginning of the file, our version, and the rest of the file.
+                            def newPomFile = pomFile.replaceFirst(parentVersionPattern, "\$1${config.cp_version}\$2")
```

Comment:
```
                            def newPomFile = pomFile.replaceFirst(parentVersionPattern, "\$1${config.cp_version}\$2")
```

Is the `\$1` groovy syntax to reference matched groups?

Reply:
yes